lib/msf/core/exploit/remote/smtp_deliver.rb

```diff
@@ -246,10 +256,16 @@ def smtp_send_recv(cmd, nsock=self.sock)
     begin
       nsock.put(cmd)
       res = nsock.get_once
-      while !(res =~ /(^|\r\n)\d{3}( .*|)\r\n$/) && chunk = nsock.get_once
-        res += chunk
+      while !(res =~ /(^|\r\n)\d{3}( .*|)\r\n$/)
+        chunk = nsock.get_once
```

Comment:
I'm somewhat concerned about the potential of a DDOS here; This kerberos branch needed similar logic for implementing timeouts correctly - https://github.com/rapid7/metasploit-framework/pull/17455